### PR TITLE
Filter: ErodeDilateMask, ErodeDilateCoordination filter stub code.

### PR DESCRIPTION
**Pull request by imikejackson**
Signed-off-by: Michael Jackson <mike.jackson@bluequartz.net>

## Naming Conventions

Naming of variables should descriptive where needed. Loop Control Variables can use `i` if warranted. Most of these conventions are enforced through the clang-tidy and clang-format configuration files. See the file `complex/docs/Code_Style_Guide.md` for a more in depth explanation.

## Filter Checklist

The help file `complex/docs/Porting_Filters.md` has documentation to help you port or write new filters. At the top is a nice checklist of items that should be noted when porting a filter.

## Unit Testing

The idea of unit testing is to test the filter for proper execution and error handling. How many variations on a unit test each filter needs is entirely dependent on what the filter is doing. Generally, the variations can fall into a few categories:

- [x] 1 Unit test to test output from the filter against known exemplar set of data
- [x] 1 Unit test to test invalid input code paths that are specific to a filter. Don't test that a DataPath does not exist since that test is already performed as part of the SelectDataArrayAction.

## Code Cleanup
- [x] No commented out code (rare exceptions to this is allowed..)
- [x] No API changes were made (or the changes have been approved)
- [x] No major design changes were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [x] Added license to new files (if any)
- [x] Added example pipelines that use the filter
- [x] Classes and methods are properly documented

### CODE REVIEW

File: src/Plugins/ComplexCore/CMakeLists.txt [hunk 1]
```diff
@@ -87,7 +87,12 @@ set(FilterList
   TriangleNormalFilter
   WriteASCIIDataFilter
   WriteBinaryDataFilter
-AlignSectionsListFilter
+  AlignSectionsListFilter
+
+
+  ErodeDilateBadDataFilter
+  ErodeDilateMaskFilter
```
Comment (nyoungbq) [inline src/Plugins/ComplexCore/CMakeLists.txt]: Not sure, but if not in alphabetical order I would reorder

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ErodeDilateCoordinationNumber.cpp [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+          }
+          coordinationNumber[voxelIndex] = coordination;
+          const int64 neighbor = neighbors[voxelIndex];
+          if(coordinationNumber[voxelIndex] >= m_InputValues->CoordinationNumber && coordinationNumber[voxelIndex] > 0)
+          {
+            /******************************************************************
+             * If this section is slow it is because we are having to use the
+             * ExecuteDataFunction<T>() in order to call "copyTuple()" because
+             * "copyTuple()" isn't in the IArray API set. Oh well.
+             */
```
Comment (nyoungbq) [inline src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ErodeDilateCoordinationNumber.cpp:149]: Is there any way we can utilize `CopyTupleUsingIndexList` from DataArrayUtilities? If so we could parallelize this speeding it up significantly.

**Review by nyoungbq — APPROVED** [review]

File: src/Plugins/ComplexCore/CMakeLists.txt (hunk 1 above)
Comment (imikejackson) [reply src/Plugins/ComplexCore/CMakeLists.txt]: Done

File: src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ErodeDilateCoordinationNumber.cpp (hunk 2 above)
Comment (imikejackson) [reply src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ErodeDilateCoordinationNumber.cpp:149]: That algorithm takes way too much memory to perform the same copying. I'll leave this in here until we find a minimally memory usage algorithm.

**Pull request merged**